api/src/main/java/com/netflix/iceberg/expressions/UnboundPredicate.java

```diff
@@ -67,11 +68,12 @@ Expression bind(Types.StructType struct) {
    */
   public Expression bind(Types.StructType struct, boolean caseSensitive) {
     Types.NestedField field;
-    if (caseSensitive) {
-      field = struct.field(ref().name());
-    } else {
-      field = struct.caseInsensitiveField(ref().name());
-    }
+    String expressionFieldPath = ref().name();
+
+    boolean isNestedFieldExp = expressionFieldPath.indexOf('.') > -1;
+
+    field = isNestedFieldExp ? findNestedField(struct, expressionFieldPath, caseSensitive) :
+      caseSensitive ? struct.field(ref().name()) : struct.caseInsensitiveField(ref().name());
```

Comment:
minor: can we reuse `expressionFieldPath` instead of obtaining `ref().name()` again?

Reply:
done